src/services/stringCompletions.ts

```diff
@@ -484,7 +485,19 @@ function getStringLiteralCompletionEntries(sourceFile: SourceFile, node: StringL
             const existing = new Set(namedImportsOrExports.elements.map(n => moduleExportNameTextEscaped(n.propertyName || n.name)));
             const uniques = exports.filter(e => e.escapedName !== InternalSymbolName.Default && !existing.has(e.escapedName));
             return { kind: StringLiteralCompletionKind.Properties, symbols: uniques, hasIndexSignature: false };
-
+        case SyntaxKind.BinaryExpression:
+            if ((parent as BinaryExpression).operatorToken.kind === SyntaxKind.InKeyword) {
+                const { left, right } = parent as BinaryExpression;
+                const leftOrRight = findAncestor(node, n => n.parent.kind === SyntaxKind.BinaryExpression);
+                const type = typeChecker.getTypeAtLocation(leftOrRight === right ? left : right);

```

Comment:
What exactly is the purpose of getting the type on the right side? Where is that tested?

Reply:
This code has to handle both:
- `key in obj`
- `obj in key`

Those new completions are trying to suggest available `key`s. So to do that we obtain "the other" type and based on it we compute `properties` (one line below this) based on which those are computed.

Reply:
What language construct is `obj in key` supposed to correspond to though? Isn't that just someone writing a construct backwards?

Reply:
That's a new construct coming in the next edition of JabaScript.

You are, **of course**, right - this was nonsense. I'm sometimes a dumdum when producing code in short 5min spans of time I have at hand.

I just pushed out a commit removing this bit.